packages/core/src/tracing/sentrySpan.ts

```
@@ -132,6 +138,17 @@ export class SentrySpan implements Span {
     return this;
   }
 
+  /** @inheritDoc */
+  public addFeatureFlag(name: string, value: boolean): this {
+    if (!(name in this._attributes)) {
+      if (this._numFeatureFlags >= MAX_FEATURE_FLAGS_PER_SPAN) {
+        return this;
+      }
```

Should this be a buffer so that we can evict the least recently added flags (similar to what we do w/ breadcrumbs)? Silently ignoring the flag here feels not great.

---

It's what we do on in python atm, cc @cmanallen. We could have a buffer but need to make sure the attribute is serializable - Doesn't look like SpanAttributeValue allows nested objects. Maybe we could store a buffer of keys internally

---

For spans the oldest flag evaluations are the most important because they're likely to have the largest performance impact.  We drop late arrivals due to request size concerns.